src/shopper.py

```diff
@@ -0,0 +1,318 @@
+            "CLAW2": [load_template("assets/shop_1280_720/claws/claw2.png", 1.0), 1.0],
+            "CLAW3": [load_template("assets/shop_1280_720/claws/claw3.png", 1.0), 1.0],
+            "TO_TRAPS": [load_template("assets/shop_1280_720/claws/to_traps.png", 1.0), 1.0],
+            "3_TO_TRAPS": [load_template("assets/shop_1280_720/claws/3_to_traps.png", 1.0), 1.0],
+            "2_TO_ASSA": [load_template("assets/shop_1280_720/claws/2_to_assa.png", 1.0), 1.0],
+            "TO_LIGHT": [load_template("assets/shop_1280_720/claws/to_light.png", 1.0), 1.0],
+            "TO_WB": [load_template("assets/shop_1280_720/claws/wb.png", 1.0), 1.0],
+            "TO_DS": [load_template("assets/shop_1280_720/claws/to_ds.png", 1.0), 1.0],
+            "TO_VENOM": [load_template("assets/shop_1280_720/claws/to_venom.png", 1.0), 1.0],
+        }
+        self._templates.update(custom_templates)
+
+
+def exit(run_obj):
+    run_time = str(datetime.timedelta(seconds=round(time.time() - run_obj.start_time)))
+    Logger.info("Exiting shopping mall...")
+    print(
+        "STATS \truns \t\ttime \t\tias_gloves_seen \tgloves_bought \tclaws_evaluated \tclaws_bought\n"
+        f"\t{run_obj.run_count} \t\t{run_time}"
+        f"\t\t{run_obj.ias_gloves_seen} \t\t\t{run_obj.gloves_bought} \t\t{run_obj.claws_evaluated} \t\t\t{run_obj.claws_bought}"
+    )
+    os._exit(0)
+
+
+def wait_for_loading_screen(screen: Screen, time_out):
+    start = time.time()
+    while time.time() - start < time_out:
+        img = screen.grab()
+        is_loading_black_roi = np.average(img[:700, 0:250]) < 4.0
+        if is_loading_black_roi:
+            return True
+    return False
+
+class AnyaShopper:
+    """
+    Shop at Anya for 20 ias, +3 java skill gloves and more...
+
+    In order to start the shopping bot:
+    1.) Run this this file in Python.
+    2.) Be ingame in Harrogath (act 5 town).
+    3.) Stand close to Anya and the red portal.
+    4.) While being ingame, press resume_key (default F11) to start the shopping, and exit_key (default F12) to stop it.
+    """
+
+    def __init__(self, config):
+        
+        # Configurable part, dont touch anything above
+        # Set look_for variables to False if you dont like your personal shopper to look for these
+        # Obviously something need to be set to True, or your shopper will be very confused
+        # For the claw scores use 7 if you are happy with any + to traps or +2 assassin, 9 if you want at least +2 assassin or two useful trap stats, 11 if you 
+        
+        self.look_for_plus_2_gloves = True
+        self.look_for_plus_3_gloves = True  
+        self.look_for_trap_claws = True
+        self.trap_claw_min_score = 11
+        self.look_for_melee_claws = True
+        self.melee_claw_min_score = 11
+        
+        # Dont touch anything below here
+        
+        self._screen = screen.Screen(config.general["monitor"])
```

Comment:
Pass the screen instance to the constructor instead of creating a new one. Otherwise two screen instances are created.

Reply:
Oh, I dont know Python enough for that - I am a pseudo coder. Can someone help?

Reply:
I can do : )